### ISSUE TITLE
Run coupled control test case using gnu compiler

### ISSUE THREAD

**Issue by MinsukJi-NOAA** [ufs-community/ufs-weather-model#569]
## Description
Run-time failure occurs when running cpld_control using the gnu compiler. Running cpld_control using gnu is helpful for many reasons, with the immediate need here being able to run cpld_control in CI test.

## Related to
#543 

## To Reproduce:
1. git clone --branch feature/utest-update --recurse-submodules https://github.com/MinsukJi-NOAA/ufs-weather-model
2. cd ufs-weather-model/test
3. export RT_COMPILER=gnu
4. ./utest -n cpld_control -c rst >out 2>&1 &

## Output

**output logs**
/scratch1/NCEPDEV/stmp2/Minsuk.Ji/FV3_UT/ut_191770/cpld_control_std_base/err

**Comment by DeniseWorthen** [comment]
I'm having trouble reproducing @junwang-noaa 's CMEPS error. It looks like her test is older (your 2a498db commit). If I try running from your current commit (b38fea1), I get an error from copying in the input (@[ATMRES]_L@[NPZ].mx@[OCNRES]_frac instead of @[ATMRES].mx@[OCNRES]_frac). I can resolve that error but then when I run, the first err message I get is 

```
116: FATAL from PE   116: check_nml_error in fms_mod:  Unknown namelist, or mistyped namelist variable in namelist atmos_model_nml, (IOSTAT =         5010 )
```

This is what I see in your err file also. I will try the older commit now to look for the CMEPS error.

**Comment by climbfuji** [comment]
I got this error yesterday. It's a bug in the input.nml, the value of ccpp_suite is a string and must be in quotes.

**Comment by MinsukJi-NOAA** [comment]
@DeniseWorthen , my error is actually from the old commit.

**Comment by DeniseWorthen** [comment]
@climbfuji There is a bug in input.nml from my commit yesterday?

**Comment by climbfuji** [comment]
No, it's been there "all the time", but Intel doesn't care.

**Comment by DeniseWorthen** [comment]
I see. So I need to fix this for the coupled tests. By adding the quotes around ccpp_suite I'm able to recover the CMEPS error.

**Comment by junwang-noaa** [comment]
Sorry, forgot to mention. I fixed the ccpp_suites file in namelist
/scratch1/NCEPDEV/stmp2/Jun.Wang/FV3_UT/ut_194438/cpld_control_std_base/input.nml

**Comment by DeniseWorthen** [comment]
@MinsukJi-NOAA Is there a way to use build.sh with gnu as I do w/ intel? If RT_compiler is set as GNU, can I do:

CMAKE_FLAGS="-DAPP=S2S" CCPP_SUITES="FV3_GFS_2017_coupled,FV3_GFS_2017_satmedmf_coupled,FV3_GFS_v15p2_coupled" ./build.sh >output 2>&1 &

**Comment by MinsukJi-NOAA** [comment]
@DeniseWorthen I have not tried it. @DusanJovic-NOAA , does build.sh work with gnu compiler? How does build.sh load modulefiles? Do we just set CMAKE_Platform=hera.gnu?

**Comment by DeniseWorthen** [comment]
I actually deleted my comment after giving it a try. It seems to have worked by loading ufs_hera.gnu and having the RT_compiler set to GNU:

-- The C compiler identification is GNU 9.2.0
-- The CXX compiler identification is GNU 9.2.0
-- The Fortran compiler identification is GNU 9.2.0

**Comment by DeniseWorthen** [comment]
I compiled in debug mode and found a bounds issue: 

```  
0: At line 778 of file /scratch2/NCEPDEV/climate/Denise.Worthen/WORK/ufs_test/CMEPS-interface/CMEPS/mediator/esmFlds.F90
0: Fortran runtime error: Index '1' of dimension 1 of array 'fldlistmed_aoflux%flds' above upper bound of 0
```

I have a fix, although I'm a bit puzzled why my first try didn't work. That fix failed because

size(fldListMed_aoflux%flds) /= fieldCount  where fieldCount=med_fldList_GetNumFlds(fldListMed_aoflux). This bothers me so I need to dig a bit. 

I was able to put in a fix using fieldCount and I was able to run the cpld_control through to the end in debug mode.

**Comment by DeniseWorthen** [comment]
Branch containing gnu compile fix https://github.com/DeniseWorthen/CMEPS/tree/bugfix/gnu_compile

**Comment by MinsukJi-NOAA** [comment]
I will give that a try. Thank you @DeniseWorthen

**Comment by DeniseWorthen** [comment]
I'm getting a different error now when I build this CMEPS branch with GNU and an otherwise up-to-date ufs-weather:

```
20210525 194727.359 ERROR            PET000 src/addon/NUOPC/src/NUOPC_Base.F90:2086 Invalid argument  - u is not a StandardName in the NUOPC_FieldDictionary!
20210525 194727.360 ERROR            PET000 src/addon/NUOPC/src/NUOPC_Base.F90:476 Invalid argument  - Passing error in return code
20210525 194727.360 ERROR            PET000 /scratch2/NCEPDEV/climate/Denise.Worthen/WORK/ufs_test_gnu/FV3/fv3_cap.F90:871 Invalid argument  - Passing error in return code
```

**Comment by junwang-noaa** [comment]
This is in debug mode, right? @DusanJovic-NOAA @mark-a-potts  do you want to change the JEDI coupled names to nuopc consistent names?

**Comment by DeniseWorthen** [comment]
I know what happened. I was using the same run directory which had an older fd_nems.yaml (from Minsuk's branch). I added the newer file with the JEDI fields and I can now run after compiling w/ gnu.

**Comment by MinsukJi-NOAA** [comment]
Fixes were implemented via #529
